src/Lykke.Service.BlockchainWallets/Controllers/WalletsController.cs

```diff
@@ -0,0 +1,211 @@
+
+            await _walletService.DeleteWalletAsync(integrationLayerId, integrationLayerAssetId, clientId);
+
+            return Accepted();
+        }
+
+        [HttpGet("by-client-ids/{clientId}/address")]
+        public async Task<IActionResult> GetAddress([FromRoute] string integrationLayerId,
+            [FromRoute] string integrationLayerAssetId, [FromRoute] Guid clientId)
+        {
+            if (!ValidateRequest(integrationLayerId, integrationLayerAssetId, clientId, out var badRequest))
+            {
+                return badRequest;
+            }
+
+            var address = await _walletService.GetAddressAsync(integrationLayerId, integrationLayerAssetId, clientId);
+
+            if (string.IsNullOrEmpty(address))
+            {
+                return NoContent();
+            }
+
+            return Ok(new AddressResponse
+            {
+                Address = address
+            });
+        }
+
+        [HttpGet("by-addresses/{address}/client-id")]
+        public async Task<IActionResult> GetClientId([FromRoute] string integrationLayerId,
+            [FromRoute] string integrationLayerAssetId, [FromRoute] string address)
+        {
+            if (!ValidateRequest(integrationLayerId, integrationLayerAssetId, address, out var badRequest))
+            {
+                return badRequest;
+            }
+
+            var clientId = await _walletService.GetClientIdAsync(integrationLayerId, integrationLayerAssetId, address);
+
+            if (!clientId.HasValue)
+            {
+                return NoContent();
+            }
+
+            return Ok(new ClientIdResponse
+            {
+                ClientId = clientId.Value
+            });
+        }
+
+        private bool ValidateRequest(string integrationLayerId, string integrationLayerAssetId, string address,
+            out IActionResult badRequest)
+        {
+            var invalidInputParams = new List<string>();
+
+            if (string.IsNullOrWhiteSpace(integrationLayerId))
+            {
+                invalidInputParams.Add(nameof(integrationLayerId));
+            }
+
+            if (string.IsNullOrWhiteSpace(integrationLayerAssetId))
+            {
+                invalidInputParams.Add(nameof(integrationLayerAssetId));
+            }
+
+            if (string.IsNullOrWhiteSpace(address))
+            {
+                invalidInputParams.Add(nameof(address));
+            }
+
+            if (!invalidInputParams.Any())
+            {
+                badRequest = null;
+
+                return true;
+            }
+
+            badRequest = BadRequest
+            (
+                ErrorResponse.Create(
+                    $"One or more input parameters [{string.Join(", ", invalidInputParams)}] are invalid.")
+            );
+
+            return false;
+        }
+
+        private bool ValidateRequest(string integrationLayerId, string integrationLayerAssetId, Guid clientId,
+            out IActionResult badRequest)
+        {
```

Comment:
Current version is ok for now, but just for information - you could use [ErrorResponse.AddModelError](https://github.com/LykkeCity/CommonDotNetLibraries/blob/master/src/Lykke.Common/Api/Contract/Responses/ErrorResponse.cs#L44) here. You code could be looks like:

```csharp
var errorResponse = new ErrorResponse();

if (string.IsNullOrWhiteSpace(integrationLayerId))
{
    errorResponse.AddModeError(nameof(integrationLayerId), "Field is required");
}

...

if (!errorResponse.ModelErrors.Any())
{
    badRequest = null;
    return true;
}

badRequest = BadRequest(errorResponse);
```